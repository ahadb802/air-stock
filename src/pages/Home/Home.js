import React, { useEffect, useState } from 'react';
import Form from 'react-bootstrap/Form';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import Cards from '../../components/Card/Cards';
import { allCoins } from '../../redux/home-reducer';

const Home = () => {
  const [search, setSearch] = useState('');
  const handleSearch = (e) => {
    setSearch(e.target.value);
  };
  const Coins = useSelector((state) => state.coin);
  const send = useDispatch();
  useEffect(() => {
    send(allCoins());
  }, [send]);
  const filter = Coins ? Coins.filter(
    (Coin) => Coin.name
      .toLowerCase()
      .includes(
        search.toLowerCase()
              || Coin.name.toLowerCase().includes(search.toLowerCase()),
      ) || Coin.rank.toString().includes(search.toString()),
  ) : [];
  return (
    <section>
      <Form className="d-flex" style={{ width: '18rem' }}>
        <Form.Control
          type="search"
          placeholder="Enter Your Coin Name"
          className="me-2"
          aria-label="Search"
          onChange={handleSearch}
          value={search}
        />
      </Form>
      <div className="section">
        {
                       filter.map((item) => (
                         <NavLink
                           to={`Details/${item.name}`}
                           key={item.id}
                         >
                           <Cards
                             name={item.name}
                             priceUsd={item.price}
                             rank={item.rank}
                             symbol={item.symbol}
                             image={item.icon}
                           />
                         </NavLink>
                       ))
                    }
      </div>
    </section>
  );
};
export default Home;
