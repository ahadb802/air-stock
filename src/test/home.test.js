import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import store from '../redux/configure';
import Home from '../pages/Home/Home';

describe('Testing page deployment', () => {
  test('Test For Home Page to check if it is rendering or not', () => {
    const output = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(output).toMatchSnapshot();
  });
});
