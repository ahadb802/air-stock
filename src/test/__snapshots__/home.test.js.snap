// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing page deployment Test For Home Page to check if it is rendering or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <form
          class="d-flex"
          style="width: 18rem;"
        >
          <input
            aria-label="Search"
            class="me-2 form-control"
            placeholder="Enter Your Coin Name"
            type="search"
            value=""
          />
        </form>
        <div
          class="section"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <form
        class="d-flex"
        style="width: 18rem;"
      >
        <input
          aria-label="Search"
          class="me-2 form-control"
          placeholder="Enter Your Coin Name"
          type="search"
          value=""
        />
      </form>
      <div
        class="section"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
